### PR TITLE
Script generator

### PR DESCRIPTION
**Pull request by at88mph**
Fairly basic script generator.  Will append auth token and expiry date when authentication is provided in the constructor.

### CODE REVIEW

**Comment by at88mph** [comment]
The assumption is that the provided expiry date will be `OIDCUtil.ID_TOKEN_EXPIRY_MINUTES`.

**Review by brianmajor — COMMENTED** [review]
Looks good to me aside from the few comments.  Only other thing is whether the IllegalArgumentExceptions are right.  They seem like the right exception for the situation, but it would be confusing to users if they got a 400 with those messages, so the caller of this class would probably need to catch those and map them to internal errors.

**Comment by at88mph** [comment]
Rework has been submitted.  Please merge if acceptable and I can keep these MRs small with the next one.

**Comment by at88mph** [comment]
Thanks @brianmajor, I did miss that, sorry.  We could use checked Exceptions there, which would force the caller to deal with it, but Storage Inventory, for example, throws an `IllegalArgumentException` (or an `IllegalStateException` on occasion) to assert not null inputs.  Do we have another common API for this?

File: cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ScriptGenerator.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    public ScriptGenerator(final Iterator<DownloadDescriptor> downloadDescriptors, final String authToken,
+                           final Date expiryDate) {
+        this.downloadDescriptors = downloadDescriptors;
+        this.authToken = authToken;
+        this.expiryDate = expiryDate;
+
+        // If the auth token was provided, the expiry date should be provided as well.
+        if (this.expiryDate == null && StringUtil.hasLength(this.authToken)) {
+            throw new IllegalArgumentException("The Expiry Date is required when the token is supplied.");
+        } else if (this.downloadDescriptors == null) {
```
[the remark this answers is not in the document]
Comment (at88mph) [reply cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ScriptGenerator.java]: Good, thanks!

**Comment by at88mph** [comment]
Replacing with full MR with Servlet changes.

**Pull request closed without merging**